app/src/main/java/com/napzak/market/presentation/registration/RegistrationScreen.kt

```diff
@@ -0,0 +1,338 @@
+import com.napzak.market.R
+import com.napzak.market.R.string.regi_title_placeholder
+import com.napzak.market.R.string.regi_topbar_title
+import com.napzak.market.core.designsystem.component.button.CommonButton
+import com.napzak.market.core.designsystem.component.topbar.CloseTopBar
+import com.napzak.market.core.designsystem.theme.NapzakMarketTheme
+import com.napzak.market.core.type.ProductConditionType
+import com.napzak.market.core.type.TradeType
+import com.napzak.market.presentation.registration.component.RegistrationBuyGroup
+import com.napzak.market.presentation.registration.component.RegistrationGenreButton
+import com.napzak.market.presentation.registration.component.RegistrationPhotoPicker
+import com.napzak.market.presentation.registration.component.RegistrationPlainTextField
+import com.napzak.market.presentation.registration.component.RegistrationSellGroup
+import com.napzak.market.presentation.registration.state.RegistrationUiState
+import com.napzak.market.presentation.registration.type.NumeralInputType
+import com.napzak.market.presentation.registration.type.PlainTextInputType
+import com.napzak.market.presentation.registration.type.PostFeeType
+
+@Composable
+fun RegistrationRoute(
+    tradeType: String,
+    navigateUp: () -> Unit,
+    navigateToGenreSearch: () -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: RegistrationViewModel = hiltViewModel(),
+) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    val currentListSize = (MAX_ITEMS - uiState.imageUrlList.size).coerceAtLeast(MIN_ITEMS)
+    val getImageStorageLauncher = rememberLauncherForActivityResult(
+        ActivityResultContracts.GetMultipleContents()
+    ) { uris: List<Uri> ->
+        val updatedListSize = uiState.imageUrlList.size + uris.size
+        if (updatedListSize <= 10) {
+            viewModel.updatePhotoList(uris.map { it.toString() })
+        } else {
+            val remainingUris = uris.take(MAX_ITEMS - uiState.imageUrlList.size)
+            viewModel.updatePhotoList(remainingUris.map { it.toString() })
+        }
+    }
+    val getPhotoPickerLauncher = when (uiState.imageUrlList.size) {
+        MAX_ITEMS - 1 -> rememberLauncherForActivityResult(
+            ActivityResultContracts.PickVisualMedia()
+        ) { uri: Uri? ->
+            uri?.let { viewModel.updatePhotoList(listOf(it.toString())) }
+        }
+        else -> rememberLauncherForActivityResult(
+            ActivityResultContracts.PickMultipleVisualMedia(maxItems = currentListSize)
+        ) { uris: List<Uri> ->
+            val updatedListSize = uiState.imageUrlList.size + uris.size
+            if (updatedListSize <= 10) {
+                viewModel.updatePhotoList(uris.map { it.toString() })
+            } else {
+                val remainingUris = uris.take(MAX_ITEMS - uiState.imageUrlList.size)
+                viewModel.updatePhotoList(remainingUris.map { it.toString() })
+            }
+        }
+    }
```

Comment:
```suggestion
    private fun handleUris(uris: List<Uri>, currentListSize: Int, updatePhotoList: (List<String>) -> Unit) {
    val updatedListSize = currentListSize + uris.size
    if (updatedListSize <= MAX_ITEMS) {
        updatePhotoList(uris.map { it.toString() })
    } else {
        val remainingUris = uris.take(MAX_ITEMS - currentListSize)
        updatePhotoList(remainingUris.map { it.toString() })
    }
}
```
p2: uri 이미지 처리 로직이 비슷한 것 같아서 이렇게 분리를 해서 재활용을 하시는 건 어떨까요??